docs/simplify-predicate.md

```diff
@@ -0,0 +1,88 @@
+# Simplifying predicates
+
+The goals of simplification are:
+
+1. Transform the predicate into [disjunctive normal form] (normalization).
+2. Apply identities to reduce the complexity of the predicate (simplification proper).
+
+Simplification happens recursively, upward from the bottom. Therefore, we
+describe the simplification steps based on the shape of the top-most
+unsimplified clause of the predicate.
```

Comment:
This phrasing is not clear for me. If we start simplifying from the bottom, wouldn't the top look the same until the end? Maybe my intuition of what is a clause is wrong.